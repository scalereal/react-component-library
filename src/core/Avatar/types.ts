
export type AvatarSizes = 'ES' | 'S' | 'M' | 'L' ;

export interface Props{
    size?: AvatarSizes;
    bgColor?:string;
    textColor?:string;
    hoverEffect?:boolean;
    hoverTextColor?:string;
    hoverBgColor?:string;
    hoverShadowColor?:string;
    hoverTextShadowColor?:string;
    res?: string;
    disabled?:boolean;
}