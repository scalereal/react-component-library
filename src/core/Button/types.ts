import { ElementType, ReactNode } from 'react';
import { HTMLProps, Omit, WithThemeProp } from '../../utils';

export type ComponentSize = 'XS' | 'S' | 'M' | 'L' | 'XL';

export type ButtonType = 'primary' | 'secondary' | 'warning' | 'danger' | 'success';

export type ButtonEdges = 'squared' | 'rounded' | 'circle';

export interface Props extends Omit<HTMLProps<HTMLDivElement>, 'size' | 'type'>, WithThemeProp {
    type?: ButtonType;
    icon?: ElementType;
    textSize?: ComponentSize;
    size?: ComponentSize;
    edges?: ButtonEdges;
    fullWidth?: boolean;
    className?: string;
    children?: ReactNode;
    loading?: boolean;
    bgColor?: string;
    textColor?: string;
    hoverEffect?: boolean;
    hoverTextColor?: string;
    hoverBgColor?: string;
    hoverShadowColor?: string;
    disabled?: boolean;
}
