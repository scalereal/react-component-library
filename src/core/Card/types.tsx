import React, { RefObject } from 'react';
import { AnyStyledComponent } from 'styled-components';

export interface WithStyle {
    Style?: AnyStyledComponent;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export type HTMLPropsReact<T> = Omit<React.HTMLProps<T>, 'ref' | 'as'>;

export interface HTMLProps<T> extends HTMLPropsReact<T> {
    ref?: ((instance: T) => void) | RefObject<T>;
}
export interface WithThemeProp {
    // eslint-disable-next-line no-undef
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
}

export type CardDirection = 'column' | 'row';
export type CardTextAlignment = 'left' | 'center' | 'right';
export interface CardProps extends HTMLProps<HTMLDivElement>, WithThemeProp {
    /** Card Image */
    src?: string;
    /** Card Title */
    title: string;
    /** Card Description */
    description: string;
    /** Card Direction */
    direction: CardDirection;
    /** Card Text Alignment */
    alignText: CardTextAlignment;
};
