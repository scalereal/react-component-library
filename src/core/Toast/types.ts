export type ToastVariants = 'success' | 'danger' | 'info' | 'warning';

export type ToastProps =  {
    /** Toast Variant */
    variant: string;
    /** Toast Title */
    title?: string;
    /** Toast message */
    message?: string;
};
