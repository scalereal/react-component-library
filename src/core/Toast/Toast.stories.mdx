import Toast, { variants, positions, borderPositions } from './Toast';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Core/Toast"
    component={Toast}
    argTypes={{
        variant: {
            control: {
                type: 'select',
                options: Object.values(variants)
            }
        },
        fixed: {
            control: {
                type: 'boolean'
            }
        },
        titleColor: {
            control: 'color'
        },
        messageColor: {
            control: 'color'
        },
        shadowColor: {
            control: 'color'
        },
        borderColor: {
            control: 'color'
        },
        position: {
            control: {
                type: 'select',
                options: Object.values(positions)
            }
        },
        borderPosition: {
            control: {
                type: 'select',
                options: Object.values(borderPositions)
            }
        }
    }}
/>

# Toast

Displays a Toast

export const Template = ({
    variant,
    title,
    message,
    titleColor,
    messageColor,
    borderColor,
    shadowColor,
    position,
    borderPosition,
    ...args
}) => {
    return (
        <Toast
            variant={variant}
            title={title}
            message={message}
            titleColor={titleColor}
            messageColor={messageColor}
            borderColor={borderColor}
            shadowColor={shadowColor}
            position={position}
            borderPosition={borderPosition}
            {...args}
        />
    );
};

<Preview>
    <Story
        name="Toast"
        args={{
            variant: 'success',
            title: 'Title',
            message: 'Lorem Ipsum is simply dummy text',
            borderPosition: 'top'
        }}
    >
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', flexWrap: 'wrap', justifyContent: 'space-around' }}>
        <Toast variant="success" title="Title" message="Lorem Ipsum is simply dummy text" titleColor="blue" messageColor="blue" />
        <Toast variant="success" title="Title" message="Lorem Ipsum is simply dummy text" titleColor="red" messageColor="red" />
    </div>
</Preview>

### Varient

There are four varient available `danger`, `warning`, `success`, `info`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexDirection: 'column' }}>
        <Toast variant="danger" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="top" />
        <Toast variant="warning" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="top" />
        <Toast variant="success" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="top" />
        <Toast variant="info" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="top" />
    </div>
</Preview>

### Border position

There are four Border position available `top` | `left` | `right` | `bottom`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexDirection: 'column' }}>
        <Toast variant="danger" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="top" />
        <Toast variant="warning" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="right" />
        <Toast variant="success" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="bottom" />
        <Toast variant="info" title="Title" message="Lorem Ipsum is simply dummy text" borderPosition="left" />
    </div>
</Preview>

### Toast position

fixed={true} `top-left` | `top-center` | `top-right` | `right-center` | `left-center` | `bottom-left` | `bottom-center` | `bottom-right`.

fixed={false} `top-left` | `top-center` | `top-right`.

<Preview withToolbar>
    <div style={{ display: 'flex', flexDirection: 'column' }}>
        <Toast variant="info" title="Title" message="Lorem Ipsum is simply dummy text" position="top-left" />
        <Toast variant="info" title="Title" message="Lorem Ipsum is simply dummy text" position="top-center" />
        <Toast variant="info" title="Title" message="Lorem Ipsum is simply dummy text" position="top-right" />
    </div>
</Preview>

### Props

<Props story="Toast" />
