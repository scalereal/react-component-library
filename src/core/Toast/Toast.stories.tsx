import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import Toast from "./Toast"
import { ToastProps } from './types'

const variants: ToastProps['variant'][] = ['danger', 'warning', 'success', 'info'];

export default {
    title: 'Components/Core/Toast',
    component: Toast,
    argTypes: {
        variant: {
            control: {
                type: 'select',
                options: Object.values(variants)
            }
        },
        titleColor: {
            control: 'color'
        },
        messageColor: {
            control: 'color'
        },
        shadowColor: {
            control: 'color'
        },
        borderColor: {
            control: 'color'
        }
    }
} as Meta;

const Template:Story = ({ variant, title, message, titleColor, messageColor, borderColor, shadowColor, ...args}) => {
    return <Toast variant={variant} title={title} message={message} titleColor={titleColor} messageColor={messageColor} borderColor=
    {borderColor} shadowColor={shadowColor} {...args} />
}

export const Default = Template.bind({});
Default.args = {
    variant:'success',
    title:'Title',
    message:'Lorem Ipsum is simply dummy text'
}
