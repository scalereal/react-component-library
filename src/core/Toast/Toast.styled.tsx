import styled from "styled-components"
import { ToastProps } from './types';
import { defaultTheme } from "../../theme/theme"

export const Toast = styled('div')<ToastProps>`
    position:relative;
    display: flex;
    flex-direction: column;
    width: 100%;
    pointer-events: auto;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid ${defaultTheme.shadowColor};
    border-radius: .3rem;
    outline: 0;
    box-shadow: 0 0.5rem 2rem ${defaultTheme.shadowColor};
`;

export const ToastHeader = styled('div')`
    display: flex;
    flex-shrink: 0;
    align-items: center;
    justify-content: space-between;
    padding: 1rem 1rem;
    border-bottom: 1px solid ${defaultTheme.shadowColor};
    border-top-left-radius: calc(.3rem - 1px);
    border-top-right-radius: calc(.3rem - 1px);
`;
export const ToastTitle = styled('div')`
    font-size:1.2em;
    color: ${defaultTheme.typographyColor};
    margin-bottom: 0;
    line-height: 1.5;
`;
export const ToastBody = styled('div')`
    position: relative;
    flex: 1 1 auto;
    padding: 1rem;
    background-color:#fff;
`;
export const ToastMessage = styled('div')`
    font-size:1em;
    color: ${defaultTheme.typographyColor};
    margin-top: 0;
    margin-bottom: 0.5rem;
`;
export const ToastFooter = styled('div')<ToastProps>`
    height:8px;
    background-color:${props => defaultTheme.statusColors[props.variant] };
`;
export const ToastCloseIcon = styled('div')`
    position:absolute;
    top:15px;
    right:10px;
    color: ${defaultTheme.iconColor};
`;
