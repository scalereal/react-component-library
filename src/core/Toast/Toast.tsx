import React,{ FC } from 'react'
import * as Styled from "./Toast.styled"
import { ToastProps,WithStyle } from './types';
import Text from '../Text/Text';

const Toast:FC<ToastProps> & WithStyle = React.memo(
    React.forwardRef((props, ref) => {
        const { variant, title, message,...restProps } = props;
        return (
            <Styled.Toast ref={ref} variant={variant} {...restProps}>
                <Styled.ToastHeader>
                    {title && <Text>{title}</Text>}
                </Styled.ToastHeader>
                <Styled.ToastBody>
                    {message && <Text>{message}</Text>}
                </Styled.ToastBody>
                <Styled.ToastFooter variant={variant} />
                <Styled.ToastCloseIcon>X</Styled.ToastCloseIcon>
            </Styled.Toast>
        );
    })
);
Toast.displayName = 'Toast';
Toast.Style = Styled.Toast;

export default Toast
