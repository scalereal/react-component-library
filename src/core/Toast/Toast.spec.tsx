import { render } from '@testing-library/react'
import * as React from 'react'
import Toast from './Toast'
import { ToastProps } from './types'

describe('Toast', () => {
    test.each(['danger', 'warning', 'success', 'info'])('should render properly with %s variant', (variant: ToastProps['variant']) => {
        const { container } = render(<Toast variant={variant} title="Title" message="Message" />);
        expect(container).toMatchSnapshot();
    });

    it('should render title and message text', () => {
        const { container } = render(<Toast variant="success" title="Title" message="Message" />);
        expect(container).toMatchSnapshot();
    });
});
