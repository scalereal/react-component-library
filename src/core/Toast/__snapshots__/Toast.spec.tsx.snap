// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should render properly with danger variant 1`] = `
<div>
  <div
    class="sc-bdfBwQ WLqQm"
  >
    <div
      class="sc-gsTCUz kdZmDy"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Title
      </span>
    </div>
    <div
      class="sc-hKgILt UimoD"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Message
      </span>
    </div>
    <div
      class="sc-jSgupP gcdruo"
    />
    <div
      class="sc-gKsewC kBjPyY"
    >
      X
    </div>
  </div>
</div>
`;

exports[`Toast should render properly with info variant 1`] = `
<div>
  <div
    class="sc-bdfBwQ WLqQm"
  >
    <div
      class="sc-gsTCUz kdZmDy"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Title
      </span>
    </div>
    <div
      class="sc-hKgILt UimoD"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Message
      </span>
    </div>
    <div
      class="sc-jSgupP oHkxX"
    />
    <div
      class="sc-gKsewC kBjPyY"
    >
      X
    </div>
  </div>
</div>
`;

exports[`Toast should render properly with success variant 1`] = `
<div>
  <div
    class="sc-bdfBwQ WLqQm"
  >
    <div
      class="sc-gsTCUz kdZmDy"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Title
      </span>
    </div>
    <div
      class="sc-hKgILt UimoD"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Message
      </span>
    </div>
    <div
      class="sc-jSgupP jMXQTc"
    />
    <div
      class="sc-gKsewC kBjPyY"
    >
      X
    </div>
  </div>
</div>
`;

exports[`Toast should render properly with warning variant 1`] = `
<div>
  <div
    class="sc-bdfBwQ WLqQm"
  >
    <div
      class="sc-gsTCUz kdZmDy"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Title
      </span>
    </div>
    <div
      class="sc-hKgILt UimoD"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Message
      </span>
    </div>
    <div
      class="sc-jSgupP kCtqeW"
    />
    <div
      class="sc-gKsewC kBjPyY"
    >
      X
    </div>
  </div>
</div>
`;

exports[`Toast should render title and message text 1`] = `
<div>
  <div
    class="sc-bdfBwQ WLqQm"
  >
    <div
      class="sc-gsTCUz kdZmDy"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Title
      </span>
    </div>
    <div
      class="sc-hKgILt UimoD"
    >
      <span
        class="sc-iBPRYJ cZbdhE"
      >
        Message
      </span>
    </div>
    <div
      class="sc-jSgupP jMXQTc"
    />
    <div
      class="sc-gKsewC kBjPyY"
    >
      X
    </div>
  </div>
</div>
`;
